src/common/src/array/data_chunk_iter.rs

```diff
@@ -237,6 +238,15 @@ impl From<RowRef<'_>> for Row {
     }
 }
 
+impl From<&Row> for CompactedRow {
+    fn from(row: &Row) -> Self {
+        let value_indices = (0..row.0.len()).collect_vec();
```

Comment:
This is heavy. Suggest accepting a `impl IntoIterator<Item = usize>` here.

Reply:
maybe we can just change the `value_indices` as `Option<&[usize]>` and None means serialize all columns. also in statetable we can check if the  `value_indices` is all columns when state table constructs and pass None later.

Reply:
Or two functions? 🤪 which is more static.

Reply:
Personally, I slightly prefer `serialize(value_indices: Option<&[usize]>)`  instead of two functions such as  `serialize` and `partial_serialize`.  Either of these looks good to me, it doesn't matter🤪.

Reply:
Should prefer `Option<&[usize]>` than `&Option<Vec<usize>>`. If I hold a `&[usize]`, then there's no zero-cost way to pass it to this function.